### PR TITLE
add readall api

### PR DESCRIPTION
**Pull request by key-er**
image

### CODE REVIEW

File: pages/api/__tests__/landingPage.test.js [hunk 1]
```diff
@@ -59,22 +54,22 @@ describe('Landing Page API Tests', () => {
...
           body: JSON.stringify(requestBody)
         });
-
-
         expect(response.status).toBe(200);
     });
+    test('should read all content successfully', async () => {
+        const response = await fetch(`${apiUrl}`, { method: "GET" });
+        expect(response.status).toBe(200);
+        // const responseBody = await response.json();
```
Comment (ajbieber) [inline pages/api/__tests__/landingPage.test.js:62]: Can we verify the content here? At least if we are doing a read all, we should verify that the content is the legnth we expect. For example if there are 2 entries in the database, we should have 

```javascript
const responseBody = await response.json();
expect(responseBody).toHaveLength(2);
```

File: pages/api/landingPage.js [hunk 2]
```diff
@@ -36,17 +36,19 @@ async function updateContent(req, res) {
   }
 }
 
-
 async function readContent(req, res) {
   var userData;
-  try {
-    if (req.query.id != null || req.query.id != '' ) {
+  try { 
+    if (req.query.id  && (req.query.id != null || req.query.id != '' )) {
```
Comment (ajbieber) [inline pages/api/landingPage.js]: Curious about this line.

The first part of the check should make sure `req.query.id` is not null and not undefined. Not sure why you need to check again for not null

**Review by ajbieber — CHANGES_REQUESTED** [review]

File: pages/api/landingPage.js (hunk 2 above)
Comment (key-er) [reply pages/api/landingPage.js]: Typo

File: pages/api/__tests__/landingPage.test.js (hunk 1 above)
Comment (key-er) [reply pages/api/__tests__/landingPage.test.js:62]: Optional check

Comment (ajbieber) [reply pages/api/__tests__/landingPage.test.js:62]: No it shouldnt be optional. Your function does 2 things, it either returns a single entry, or it returns multiple depending on your input. It always returns 200 regardless. You need to ensure that the test verifies the correct logic is getting called. With this test as it stands, this 200 response could be from either conditional, you need to verify its not the case

**Review by ajbieber — APPROVED** [review]

File: pages/api/__tests__/landingPage.test.js (hunk 1 above)
Comment (key-er) [reply pages/api/__tests__/landingPage.test.js:62]: >Can we verify the content here? At least if we are doing a read all, we should verify that the content is the legnth we expect. For example if there are 2 entries in the database, we should have

please complete sentence...

Comment (ajbieber) [reply pages/api/__tests__/landingPage.test.js:62]: Can you explain what you mean by "please complete sentence". 

My assumption is that you didn't see the code block that was under the text, but please correct me if I'm wrong

**Pull request merged**